### ISSUE TITLE
allow mbis_volume_ratios from scf_properties

### ISSUE THREAD

**Issue by loriab** [psi4/psi4#2299]
The MBIS regression reported in #2272 was fixed via #2273, but there's still a few items to tackle. This part needn't be backported.

- [ ] oeprop(wfn, "MBIS_VOLUME_RATIOS") should be fine (indep oeprop fn), but set scf_properties mbis_volume_ratios; energy("scf") will still fail as MBIS fails via QCEngine #2272 reported because those are OEProp class instantiations called from proc.py, and the free atom volumes aren't available. The oeprop.cc code could exit gracefully when free atom volumes aren't available, but having different properties lists for the two calling routes isn't good.
- [ ] so why not add the loc from oeprop() to OEProps in proc.py so that atom volumes are available? nice thought, but the fn that produces them itself calls oeprop() and energy() and descends into endless recursion. I haven't sought the logic that makes this all work together.
- [ ] add tests.

**Comment by JonathonMisiewicz** [comment]
These comments are too terse for me to follow. If you'll be taking this issue yourself and therefore it doesn't matter if others can understand (as I expect), that's great, but please assign this issue to yourself. Otherwise, can you provide additional context and a minimal not-working example?

**Comment by loriab** [comment]
Good point -- @jeffschriber understands the issue and plans to address it sometime. The summary is that the new MBIS volume ratio functionality is only half available. The `energy(); oeprop(..., "MBIS_VOLUME_RATIOS")` route is fine, but `set scf_properties [mbis_volume_ratios]; energy()` fails b/c some necessary data (free atom volumes) isn't available. This isn't a regression. It'd just be nice to even up the capabilities, esp. as a qcschema run can only use the second form.

**Comment by jeffschriber** [comment]
Exactly. I started looking into this a little while ago, but it fell off the radar a bit. I'll be able to pick it back up early next week

**Comment by jthorton** [comment]
@loriab @jeffschriber Hi  is there any update on this issue as it is breaking a workflow I have where I calculate MBIS charges using wavefunctions stored in QCArchive the attached script shows how this works and results in the following error when using psi4-1.4.1.
```
Properties will be evaluated at   0.000000,   0.000000,   0.000000 [a0]
  Computing free-atom volumes
Traceback (most recent call last):
  File "charges_from_wavefunction.py", line 72, in <module>
    psi4.oeprop(hf_wfn, "MBIS_CHARGES")
  File "/Users/joshua/mambaforge/envs/test_psi4/lib/python3.7/site-packages/psi4/driver/p4util/util.py", line 72, in oeprop
    free_atom_volumes(wfn)
    
  File "/Users/joshua/mambaforge/envs/test_psi4/lib/python3.7/site-packages/psi4/driver/p4util/prop_util.py", line 54, in free_atom_volumes
    current_en = wfn.scalar_variable('CURRENT ENERGY')
RuntimeError: 
Fatal Error: Wavefunction::scalar_variable: Requested variable CURRENT ENERGY was not set!

Error occurred in file: /Users/github/builds/conda-builds/psi4-multiout_1633712094436/work/psi4/src/psi4/libmints/wavefunction.cc on line: 1335
The most recent 5 function calls were:
```

[charges_from_wavefunction.txt](https://github.com/psi4/psi4/files/7624873/charges_from_wavefunction.txt)

**Comment by jeffschriber** [comment]
Hm, this is a bit odd. The failing function (`free_atom_volumes(wfn)`) really shouldn't be called since you're only requesting the charges. I've created a PR to fix this, in addition to fixing the larger issue with computing MBIS volume ratios via scf_properties. Interestingly, this code fails in your case since you pass a wavefunction to oeprop  that doesn't have 'CURRENT ENERGY' as a set variable. My PR doesn't address that, though I could help out if you also want to compute the volume ratios.

**Comment by jthorton** [comment]
@jeffschriber Thanks for the quick fix this unblocks a lot of work! and mbis charges should be fine for now thanks.

**Comment by loriab** [comment]
closed by #2370
